/*
 * http://github.com/mitchweaver/suckless
 *
 * This is the official "noborder" patch that prevents
 * drawing of borders if the client is the only one on the tag.
 *
 * However, on it there is a bug where it causes client to
 * grow in size indefinitely when you resizemouse().
 *
 * My version of the patch has this fixed, only doing the client growth
 * if the current tag is in "tiled" mode.
 *
 */
--- dwm/dwm.c
+++ dwm/dwm.c
@@ -1282,6 +1282,13 @@ resizeclient(Client *c, int x, int y, int w, int h)
 	c->oldw = c->w; c->w = wc.width = w;
 	c->oldh = c->h; c->h = wc.height = h;
 	wc.border_width = c->bw;
+	if (c->mon->lt[c->mon->sellt] == &layouts[0]
+	    && !c->isfullscreen) {
+		c->w = wc.width += c->bw * 2;
+		c->h = wc.height += c->bw * 2;
+		wc.border_width = 0;
+	}
+
 	XConfigureWindow(dpy, c->win, CWX|CWY|CWWidth|CWHeight|CWBorderWidth, &wc);
 	configure(c);
 	XSync(dpy, False);
